src/interactive/documentation.ts

```diff
@@ -0,0 +1,227 @@
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { withLanguageClient } from '../extension'
+import { getVersionedParamsAtPosition, setContext } from '../utils'
+
+const viewType = 'JuliaDocumentationBrowser'
+const panelActiveContextKey = 'juliaDocumentationPaneActive'
+let extensionPath: string | undefined = undefined
+let panel: vscode.WebviewPanel = undefined
+let messageSubscription: vscode.Disposable = undefined
+
+const backStack = Array<string>() // also keep current page
+let forwardStack = Array<string>()
+
+export function activate(context: vscode.ExtensionContext) {
+    // assets path
+    extensionPath = context.extensionPath
+    context.subscriptions.push(
+        vscode.commands.registerCommand('language-julia.show-documentation-pane', showDocumentationPane),
+        vscode.commands.registerCommand('language-julia.show-documentation', showDocumentation),
+        vscode.commands.registerCommand('language-julia.browse-back-documentation', browseBack),
+        vscode.commands.registerCommand('language-julia.browse-forward-documentation', browseForward),
+    )
+    setPanelContext()
+    vscode.window.registerWebviewPanelSerializer(viewType, new DocumentationPaneSerializer())
+}
+
+function showDocumentationPane() {
+    if (panel === undefined) {
+        panel = createDocumentationPanel()
+        setPanelSubscription(panel)
+    }
+    if (panel !== undefined && !panel.visible) {
+        panel.reveal()
+    }
+}
+
+function createDocumentationPanel() {
+    return vscode.window.createWebviewPanel(viewType, 'Julia Documentation Pane',
+        {
+            preserveFocus: true,
+            viewColumn: vscode.ViewColumn.Beside,
+        },
+        {
+            enableFindWidget: true,
+            // retainContextWhenHidden: true, // comment in if loading is slow, while there would be high memory overhead
+            enableScripts: true,
+        }
+    )
+}
+
+class DocumentationPaneSerializer implements vscode.WebviewPanelSerializer {
+    async deserializeWebviewPanel(deserializedPanel: vscode.WebviewPanel, state: any) {
+        panel = deserializedPanel
+        setPanelSubscription(panel)
+        const { inner } = state
+        const html = createWebviewHTML(inner)
+        _setHTML(html)
+    }
+}
+
+function setPanelSubscription(panel: vscode.WebviewPanel) {
+    panel.onDidChangeViewState(({ webviewPanel }) => {
+        setPanelContext(webviewPanel.active)
+    })
+    panel.onDidDispose(() => {
+        setPanelContext(false)
+        if (messageSubscription !== undefined) {
+            messageSubscription.dispose()
+        }
+        panel = null
```

Comment:
This will set the function argument, but you probably intent to set the global? Probably best to follow the `g_` convention for global variables for now in this codebase until we move to classes to prevent that kind of bug.

Reply:
ah, yes, good catch :)

Reply:
addressed, I didn't include `g_` convention just to avoid a conflict with https://github.com/julia-vscode/julia-vscode/pull/1438.
Maybe later I will rename it into `g_panel`